src/tools/wasm-reduce.cpp

```diff
@@ -344,10 +344,14 @@ struct Reducer
 
   void loadWorking() {
     module = make_unique<Module>();
-    Module wasm;
     ModuleReader reader;
     reader.read(working, *module);
-    wasm.features = FeatureSet::All;
+    // If there is no features section, assume we may need them all (without
+    // this, a module with no features section but that uses e.g. atomics and
+    // bulk memory would not work).
+    if (!module->hasFeaturesSection) {
+      module->features = FeatureSet::All;
+    }
```

Comment:
Do we need this, given the default behavior in #3492?

Reply:
Yes, these are separate fixes. That PR adds flags for the commands issued by the reducer (when it invokes wasm-opt to run entire sets of optimization passes). This PR is in the reducer's internals itself (where it does "destructive reduction" of replacing bits and pieces).